cylc/flow/task_pool.py

```diff
@@ -362,20 +362,35 @@ def compute_runahead(self, force=False) -> bool:
             # Find the earliest point with unfinished tasks.
             for point, itasks in sorted(self.get_tasks_by_point().items()):
                 if (
-                    points  # got the limit already so this point too
```

Comment:
This could never be true as points is populated on a different branch.

Reply:
No, it's also populated on this branch too, within the same loop. So the logic says to automatically append the next points once we have added the first one based on the state checks. (However, I'm not sure why it wasn't just stopping at the first one, since then we just take the min anyway).

Reply:
Since we're iterating in sorted order the earlier cycles should come first, so a `break` should make sense?

Reply:
It's using points later in ways which don't cause test failures, so I'll restore the original to avoid breakages.